dbms/src/IO/Compression/EncodingUtil.cpp

```diff
@@ -21,6 +21,147 @@
 namespace DB::Compression
 {
 
+template <std::integral T>
+char * writeSameValueMultipleTime(T value, UInt32 count, char * dest)
+{
+    if (unlikely(count == 0))
+        return dest;
+    if constexpr (sizeof(T) == 1)
+    {
+        memset(dest, value, count);
+        dest += count;
+    }
+    else
+    {
+        UInt32 j = 0;
```

Comment:
Seems we can directly return here and remove the following ident of else
```suggestion
    if constexpr (sizeof(T) == 1)
    {
        memset(dest, value, count);
        dest += count;
        return dest;
    }
    UInt32 j = 0;
    // ... other following codes
    // ...
```